File: pets-location/src/main/java/com/pets/domain/model/Pet.java
package com.pets.domain.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Pet {
    private Long id;
    private String name;
    private String species;
    private String breed;
    private LocalDate birthDate;
    private UUID userId;
}

File: pets_location_app/lib/data/models/pet.dart
class Pet {
  final int id;
  final String name;
  final String species;
  final String breed;
  final DateTime birthDate;
  final String? userId;
  bool hasDevice;

  Pet({
    required this.id,
    required this.name,
    required this.species,
    required this.breed,
    required this.birthDate,
    required this.userId,
    this.hasDevice = false,
  });

  factory Pet.fromJson(Map<String, dynamic> json) {
    return Pet(
      id: json['id'],
      name: json['name'],
      species: json['species'],
      breed: json['breed'],
      birthDate: DateTime.parse(json['birthDate']),
      userId: json['userId'],
    );
  }
}
